uniffi_bindgen/src/bindings/gecko/templates/WebIDLTemplate.webidl

```diff
@@ -0,0 +1,67 @@
+// This file was autogenerated by some hot garbage in the `uniffi` crate.
+// Trust me, you don't want to mess with it!
+
+{%- for rec in ci.iter_record_definitions() %}
+dictionary {{ rec.name()|class_name_webidl }} {
+  {%- for field in rec.fields() %}
+  required {{ field.type_()|type_webidl }} {{ field.name()|var_name_webidl }};
+  {%- endfor %}
+};
+{% endfor %}
+
+{%- for e in ci.iter_enum_definitions() %}
+enum {{ e.name()|class_name_webidl }} {
+  {% for variant in e.variants() %}
+  "{{ variant|enum_variant_webidl }}"{%- if !loop.last %}, {% endif %}
+  {% endfor %}
+};
+{% endfor %}
+
+{%- let functions = ci.iter_function_definitions() %}
+{%- if !functions.is_empty() %}
+[ChromeOnly, Exposed=Window]
+namespace {{ ci.namespace()|class_name_webidl }} {
+  {#-
+  // We'll need to figure out how to handle async methods. One option is
+  // to declare them as `async foo()`, or an `[Async]` or `[BackgroundThread]`
+  // attribute in the UniFFI IDL. Kotlin, Swift, and Python can ignore that
```

Comment:
Kotlin might like to automatically dispatch such methods via coroutine or similar, I think this could be worthwhile in more contexts than just Gecko.

Reply:
Ooooh, yes, great point! And maybe Swift can use a dispatch queue internally, and Python could use `asyncio`! 🚀

For the Nimbus pure-C++ API, we can punt on figuring out the exact details, and leave it up to a wrapper to initialize it and make network requests on a background task queue, just like we do in a-c and Swift now. But it’s definitely worth exploring this, and we’ll certainly want to have a story here before we transition the Nimbus JS SDK to Rust.